src/AccessControl/Implementation.py

```diff
@@ -25,6 +25,13 @@
 
 """
 from __future__ import absolute_import
+import os
+
+# We cannot use `PURE_PYTHON` as this would be propagated to `ExtensionClass`
+# and `Acquisition`. Due to restrictions of python implementation concerning
+# proxies it is not possible to run all with pure python and we influence only
+# the AccessControl implementation here.
+CAPI = not int(os.environ.get('AC_PURE_PYTHON', '0'))
```

Comment:
This should also honor `PURE_PYTHON`, i.e. CAPI = not AC_PURE_PYTHON and not PURE_PYTHON

Reply:
As written in the comment above, it is not possible to have all packages `AccessControl`, `Acquisition` and `ExtensionClass`. The `CAPI` is also only valid in this package, so I do not see a benefit to include the switch of the other packages.